### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by yegonkimutai**
This PR, my partner and I,
- [ ] Add tests for the different classes: `Book`, `Person`, `Student`, `Teacher`, `Classroom`, `Rental`
- [ ] Add tests for the decorator methods: `CapitalizeDecorator`, `TrimmerDecorator`

### CODE REVIEW

File: spec/persons_spec.rb
Comment (brytebee) [inline spec/persons_spec.rb:1]: Great job 👏👏
Well-written tests and they all passed.

However, all methods defined in all classes should be tested. For instance, this `Person` class should test the methods:
 - correct_name
 - add_rental

https://github.com/yegonkimutai/OOP-school-library/blob/b2f3a3e782722bc31f92e01e3392ca427940e933/persons.rb#L1-L34

Ensure to take a sweep through all other classes and ensure there is a test for every method, especially the public methods.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @yegonkimutai & @AbbasSarwar 👋,

##

### Highlights💡
- Good use of the ruby test suit.✔️
- Well-written tests and they all passed.✔️
- Good commit message.✔️ 
- Good PR Summary.✔️
- App functions as it should.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/persons_spec.rb
Comment (AbbasSarwar) [reply spec/persons_spec.rb:1]: added methods

**Review by omarsalem7 — APPROVED** [review]
Hi @yegonkimutai, @AbbasSarwar 😊, 
# STATUS: APPROVED! 🥇 💯 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ All unit tests passed
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**